server/src/lib/utils/RequestUrlGetter.ts

```diff
@@ -0,0 +1,23 @@
+import Constants = require("../../../../shared/constants");
+import Express = require("express");
+import GetHeader from "./GetHeader";
+import HasHeader from "./HasHeader";
+
+export class RequestUrlGetter {
+  static getOriginalUrl(req: Express.Request): string {
+
+    if (HasHeader(req, Constants.HEADER_X_ORIGINAL_URL)) {
+      return GetHeader(req, Constants.HEADER_X_ORIGINAL_URL);
+    }
+
+    const proto = GetHeader(req, Constants.HEADER_X_FORWARDED_PROTO);
+    const host = GetHeader(req, Constants.HEADER_X_FORWARDED_HOST);
```

Comment:
Any reason not to include the port as @ViViDboarder did? https://github.com/clems4ever/authelia/blob/9304e834162b8f237a08367b08d95477ef2ed035/server/src/lib/utils/RequestUrlGetter.ts#L15

Note that this could very well differ from 80 and 443 in some instances.

Reply:
Well in my current setup X-Forwarded-Host embeds the port but I guess I misconfigured nginx. I will have a look.

Reply:
Please see my answer in below comment. X-Fowarded-Host hold the domain AND the port as you can see in the following log:
```debug: date='Thu Apr 11 2019 21:52:11 GMT+0200 (CEST)' method='GET', path='/api/verify' requestId='cc6e2002-df4b-4774-9d24-eee7a708648c' sessionId='zpB5cJqbdABqeKC9fhnUjvi3CnlEG_Sm' ip='192.168.240.1' message='Header x-forwarded-host is set to secure.example.com:8080'```
And in my setup Traefik is not on port 443 or 80 but 8080 instead.

Reply:
Thanks @clems4ever, just wanted to be sure 😄